src/Storages/AlterCommands.cpp

```diff
@@ -74,8 +75,11 @@ std::optional<AlterCommand> AlterCommand::parse(const ASTAlterCommand * command_
         }
 
         if (ast_col_decl.codec)
+        {
+            if (ast_col_decl.default_specifier == "ALIAS")
+                throw Exception{ "Cannot specify codec for column type ALIAS", ErrorCodes::ILLEGAL_SYNTAX_FOR_DATA_TYPE};
```

Comment:
Actually `ALIAS` is not a data type :) Maybe `BAD_ARGUMENTS` would be better here, but I'm not sure.

Reply:
sure, that sounds better, I have made the change.